static/src/javascripts/projects/common/utils/fastdom-promise.js

```diff
@@ -0,0 +1,34 @@
+define([
+    'common/utils/_',
+    'fastdom',
+    'Promise'
+], function (
+    _,
+    fastdom,
+    Promise
+) {
+    return {
+        read: function (fn, context) {
+            return new Promise(function (resolve, reject) {
+                fastdom.read(function () {
+                    if (_.isFunction(fn)) {
+                        resolve(fn.call(context));
+                    } else {
+                        reject();
+                    }
+                });
+            });
+        },
+        write: function (fn, context) {
```

Comment:
agree with @robertberry .

I think it's nicer to write
```
read: function (fn, context) {
    if (_.isFunction(fn)) {
        throw new TypeError('....');
    }
    return new Promise(function (resolve) {
        fastdom.read(function () {
            resolve(fn.call(context));
        });
    });
},
```

because `fn` is mandatory and `rejecting` will basically make the error silent.